type-check.ts

```diff
@@ -201,10 +231,39 @@ export function tcStmt(env : GlobalTypeEnv, locals : LocalTypeEnv, stmt : Stmt<S
       return {a: tRet.a, tag: stmt.tag, value:tRet};
     case "while":
       var tCond = tcExpr(env, locals, stmt.cond);
+      locals.whileCount = locals.whileCount+1;
+      locals.currLoop.push(["while", locals.whileCount]);
       const tBody = tcBlock(env, locals, stmt.body);
+      locals.currLoop.pop();
       if (!equalType(tCond.a[0], BOOL)) 
         throw new TypeCheckError("Condition Expression Must be a bool");
       return {a: [NONE, stmt.a], tag:stmt.tag, cond: tCond, body: tBody};
+    case "for":
+      var tVars = tcExpr(env, locals, stmt.vars);
+      var tIterable = tcExpr(env, locals, stmt.iterable);
+      locals.forCount = locals.forCount+1;
+      locals.currLoop.push(["for",locals.forCount]);
+      var tForBody = tcBlock(env, locals, stmt.body);
+      locals.currLoop.pop();
+      if(!equalType(tVars.a[0], NUM))
+        throw new TypeCheckError("Expected type `int`, got type `" + tVars.a[0].tag + "`");
+      if(tIterable.a[0].tag !== "class" || tIterable.a[0].name !== "range")
+        throw new TypeCheckError("Not an iterable");
+      if(stmt.elseBody !== undefined) {
+        const tElseBody = tcBlock(env, locals, stmt.elseBody);
+        return {a: [NONE, stmt.a], tag: stmt.tag, vars: tVars, iterable: tIterable, body: tForBody, elseBody: tElseBody};
+      }
+      return {a: [NONE, stmt.a], tag: stmt.tag, vars: tVars, iterable: tIterable, body: tForBody};
+    case "break":
+      if(locals.currLoop.length === 0)
```

Comment:
loopDepth is not needed for typechecking. It is an optional argument that is getting p[opulated so that lower.ts can use it to understand which corresponding loop label to jump to (say the 10th for (end/start/body) or the 2nd while  (end/start/body)). We could have done this computation in lower.ts, but made the decision to populate any optional arguments in the typechecker itself